tests/test_utils_http.py

```diff
@@ -332,3 +329,32 @@ def test_adjust_range_header():
         _adjust_range_header("bytes=0-100", 150)
     with pytest.raises(RuntimeError):
         _adjust_range_header("bytes=-50", 100)
+
+
+def test_proxy_env_is_used(monkeypatch):
+    """Regression test for https://github.com/huggingface/transformers/issues/41301.
+
+    Test is hacky and uses httpx internal attributes, but it works.
+    We just need to test that proxies from env vars are used when creating the client.
+    """
+    monkeypatch.setenv("HTTP_PROXY", "http://proxy.example1.com:8080")
+    monkeypatch.setenv("HTTPS_PROXY", "http://proxy.example2.com:8181")
+
+    client = get_session()
```

Comment:
the test is currently failing because the global `httpx.Client` is cached by `get_session()` and earlier tests create the client before `HTTP_PROXY`/`HTTPS_PROXY` are set, so the cached client has no proxies mounted.
we should recreate the client after setting the env vars
```suggestion
    set_client_factory(default_client_factory)
    client = get_session()
```